components/butterfly/src/server/mod.rs

```diff
@@ -725,21 +726,18 @@ impl Server {
     pub fn service_config_for(&self,
                               service_group: &str,
                               incarnation: Option<u64>)
-                              -> Option<(u64, String)> {
+                              -> Option<(u64, toml::Value)> {
         let mut result = None;
-        self.service_config_store
-            .with_rumor(service_group, "service_config", |maybe_sc| {
-                if let Some(sc) = maybe_sc {
-                    if incarnation.is_none() || sc.get_incarnation() > incarnation.unwrap() {
-                        match sc.config() {
-                            Ok(config) => result = Some((sc.get_incarnation(), config)),
-                            Err(e) => {
-                                warn!("Cannot decrypt service config for {}: {}", service_group, e)
-                            }
-                        }
+        self.service_config_store.with_rumor(service_group, "service_config", |maybe_sc| {
+            if let Some(sc) = maybe_sc {
+                if incarnation.is_none() || sc.get_incarnation() > incarnation.unwrap() {
+                    match sc.config() {
+                        Ok(config) => result = Some((sc.get_incarnation(), config)),
+                        Err(err) => warn!("{}", err),
```

Comment:
The error returned has a much better message associated with it now. In Rust, this syntax will call the fmt trait implementation of the thing it's replacing. So that's where the improved error is. We don't need any additional context anymore here